wiring/src/spark_wiring_cloud.cpp

```diff
@@ -76,3 +76,9 @@ Future<bool> CloudClass::publish_event(const char *eventName, const char *eventD
 int CloudClass::publishVitals(system_tick_t period_s_) {
     return spark_publish_vitals(period_s_, nullptr);
 }
+
+void CloudClass::cancel() {
```

Comment:
I would hold off adding a new public API for a bugfix release and instead give the customer this code block as an example of how to synchronously cancel the cloud connection. Let's also document the expected behavior, specifically that the ongoing connection isn't actively interrupted, but will continue to the next logical step. E.g. if the handshake is ongoing then the handshake will continue to completion or failure, but will not be prematurely cancelled.

Reply:
I renamed this method to `Particle._cancel()` as we discussed.